functional_test/pet_store_client.py

```diff
@@ -9,6 +9,17 @@
 from urlparse import parse_qs
 from urlparse import urlsplit
 
+def map_to_query_params(m, *args):
```

Comment:
I have not tested this, but it appears as though this kind of mapping is already present in the python `requests` library and it is possible I have been doing it all wrong?

The following is from the requests documentation.  Seemingly you can do `self.session.request(..., params=your_dictionary_here)`.  Note I didn't try it yet

```
        :param params: (optional) Dictionary or bytes to be sent in the query
            string for the :class:`Request`.
```

Reply:
Just confirmed this does work.  I think we should use this method, and perhaps a different PR go back and fix all the other methods...

```python
    def list_pets(self, page_size=10, offset=0):
        """
        Returns a list of pets
        :return:
        """
        url = urljoin(self.index_url, "/pets")
        return self.session.request(method='GET', url=url, headers=self.headers, params={'pageSize': page_size, 'offset': offset})
```

Reply:
Hey @Phill101 I added another commit to your PR that does what Paul is talking about here. Just merge that and this should be resolved. https://github.com/Phill101/scala-pet-store/pull/2 

Only problem is the way I've done it the caller of the client method can pass any parameters and they'll be put into the query string. I agree with Paul tho let's revise all these methods for parameter usage in one later PR.